### PR TITLE
DB-10472 Restrict sys visibility to only splicedb.splice - Other db owners can see everything in their db

### PR DESCRIPTION
**Pull request by arnaud-splice**

### CODE REVIEW

File: mem_sql/pom.xml [hunk 1]
```diff
@@ -259,12 +259,10 @@
                             <key>com.splicemachine.enableLegacyAsserts</key>
                             <value>true</value>
                         </systemProperty>
-                        <!-- Uncomment the following to dump activation class files for mem platform to the mem_sql directory.
                            <systemProperty>
```
Comment (dgomezferro) [inline mem_sql/pom.xml:262]: Comment out again

**Review by dgomezferro — APPROVED** [review]

File: mem_sql/pom.xml (hunk 1 above)
Comment (arnaud-splice) [reply mem_sql/pom.xml]: Thanks for catching that! Done.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java [hunk 2]
```diff
@@ -1054,8 +1083,8 @@ public SchemaDescriptor getSchemaDescriptor(UUID dbId, String schemaName,
                 break;
             case SchemaDescriptor.IBM_SYSTEM_ADM_SCHEMA_NAME:
                 return sysIBMADMSchemaDesc;
-            case SchemaDescriptor.STD_SYSTEM_UTIL_SCHEMA_NAME:
-                return getSystemUtilSchemaDescriptor();
+            //case SchemaDescriptor.STD_SYSTEM_UTIL_SCHEMA_NAME:
+            //    return getSystemUtilSchemaDescriptor();
```
Comment (msirek) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java:1087]: Should the commented code be removed?

**Review by msirek — APPROVED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java (hunk 2 above)
Comment (arnaud-splice) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java:1087]: done

**Pull request merged**